casadm/statistics_model.c

```diff
@@ -220,238 +217,180 @@ static void print_usage_header(FILE* outfile)
-	print_req_section(req_stats, "Write", outfile, total_reqs);
-
-	/* Pass-Through requests. */
-	percent = percentage(exp_obj_stats->read_reqs.pass_through, total_reqs);
-	print_val_perc_table_section(outfile, "Pass-Through reads", UNIT_REQUESTS,
-				     percent, "%lu",
-				     exp_obj_stats->read_reqs.pass_through);
-
-	percent = percentage(exp_obj_stats->write_reqs.pass_through, total_reqs);
-	print_val_perc_table_row(outfile, "Pass-Through writes", UNIT_REQUESTS,
-				 percent, "%lu",
-				 exp_obj_stats->write_reqs.pass_through);
-
-	/* Summary. */
-	percent = percentage(serv_reqs, total_reqs);
-	print_val_perc_table_row(outfile, "Serviced requests", UNIT_REQUESTS,
-				 percent, "%lu", serv_reqs);
-
-	print_val_perc_table_section(outfile, "Total requests", UNIT_REQUESTS,
-				     total_reqs ? 100.0f : 0.0f, "%lu",
-				     total_reqs);
-}
-
-static void print_block_section(const struct ocf_stats_block *stats_4k,
-				const char *dev_name, FILE *outfile,
-				ocf_cache_line_size_t cache_line_size)
+	print_val_perc_table_section(outfile, "Read hits",
+				     UNIT_REQUESTS, stats->rd_hits.fraction, "%lu",
+					 stats->rd_hits.value);
+	print_val_perc_table_row(outfile, "Read partial misses",
+				     UNIT_REQUESTS, stats->rd_partial_misses.fraction, "%lu",
+					 stats->rd_partial_misses.value);
+	print_val_perc_table_row(outfile, "Read full misses",
+				     UNIT_REQUESTS, stats->rd_full_misses.fraction, "%lu",
+					 stats->rd_full_misses.value);
+	print_val_perc_table_row(outfile, "Read total",
+				     UNIT_REQUESTS, stats->rd_total.fraction, "%lu",
+					 stats->rd_total.value);
+
+	print_val_perc_table_section(outfile, "Write hits",
+				     UNIT_REQUESTS, stats->wr_hits.fraction, "%lu",
+					 stats->wr_hits.value);
+	print_val_perc_table_row(outfile, "Write partial misses",
+				     UNIT_REQUESTS, stats->wr_partial_misses.fraction, "%lu",
+					 stats->wr_partial_misses.value);
+	print_val_perc_table_row(outfile, "Write full misses",
+				     UNIT_REQUESTS, stats->wr_full_misses.fraction, "%lu",
+					 stats->wr_full_misses.value);
+	print_val_perc_table_row(outfile, "Write total",
+				     UNIT_REQUESTS, stats->wr_total.fraction, "%lu",
+					 stats->wr_total.value);
+
+	print_val_perc_table_section(outfile, "Pass-Through reads",
+				     UNIT_REQUESTS, stats->rd_pt.fraction, "%lu",
+					 stats->rd_pt.value);
+	print_val_perc_table_row(outfile, "Pass-Through writes",
+				     UNIT_REQUESTS, stats->wr_pt.fraction, "%lu",
+					 stats->wr_pt.value);
+	print_val_perc_table_row(outfile, "Serviced requests",
+				     UNIT_REQUESTS, stats->serviced.fraction, "%lu",
+					 stats->serviced.value);
+
+	print_val_perc_table_section(outfile, "Total requests",
+				     UNIT_REQUESTS, stats->total.fraction, "%lu",
+					 stats->total.value);
+}
+
+#define get_stat_name(__dst, __len, __name, __postfix) \
+	memset(stat_name, 0, max_stat_len); \
+	strcpy(stat_name, __name); \
+	strcat(stat_name, postfix);
```

Comment:
Please use safe functions like `strlcpy` and `strlcat`, especially since you have length handy